src/Nethermind/Nethermind.Network.Dns/EnrDiscovery.cs

```diff
@@ -59,20 +72,40 @@ private void SearchNode(IDnsClient client, string query, SearchContext searchCon
 
         searchContext.VisitedRefs.Add(query);
 
-        string[][] lookupResult = client.Lookup(query);
+        string[][] lookupResult = await client.Lookup(query);
         foreach (string[] strings in lookupResult)
         {
             string s = string.Join("", strings);
             EnrTreeNode treeNode = EnrTreeParser.ParseNode(s);
             foreach (string link in treeNode.Links)
             {
                 DnsClient linkedTreeLookup = new(link);
-                SearchTree(linkedTreeLookup, searchContext);
+                await SearchTree(linkedTreeLookup, searchContext);
             }
 
-            foreach (string nodeRecord in treeNode.Records)
+            foreach (string nodeRecordText in treeNode.Records)
             {
-                searchContext.DiscoveredNodes.Add(nodeRecord);
+                try
+                {
+                    string broken = nodeRecordText[4..].Replace("-", "+").Replace("_", "/");
+                    broken = broken.PadRight(nodeRecordText.Length % 4);
+                    RlpStream rlpStream = new(Convert.FromBase64String(broken));
```

Comment:
This looks weird, and potentially those string.Replaces could be done in-place on char span.